registry/v2/registry.go

```diff
@@ -67,34 +68,44 @@ func (r *v2registry) Images(org string) ([]string, error) {
 	return images, err
 }
 
-func (r *v2registry) ImagesWithManifests(org string) ([]objects.Image, error) {
-	imageNames, err := r.Images(org)
+func (r *v2registry) ImagesWithManifests() ([]objects.Image, error) {
+	imageNames, err := r.Images()
 
 	if err != nil {
 		return nil, err
 	}
 
 	images := []objects.Image{}
 
-	url := strings.Replace(r.r.URL, "http://", "", 1)
-	url = strings.Replace(url, "https://", "", 1)
-	username := r.Username
-	password := r.Password
-
 	for _, imgstr := range imageNames {
-		manifest := ""
-		//TODO: find better, lightweight way to get manifest on low side
-		imageName, err := util.DockerPull(imgstr, url, org, username, password)
-		if err == nil {
-			manifest, err = util.GetSeedManifestFromImage(imageName)
+		temp := strings.Split(imgstr, ":")
+		if len(temp) != 2 {
+			r.Print("ERROR: Invalid seed name: %s. Unable to split into name/tag pair\n", imgstr)
+			continue
 		}
+		manifest, err := r.GetImageManifest(temp[0], temp[1])
 		if err != nil {
-			r.Print("ERROR: Could not get manifest: %s\n", err.Error())
+			//skip images with empty manifests
+			r.Print("ERROR: Error reading v2 manifest for %s: %s\n Skipping.\n", imgstr, err.Error())
+			continue
 		}
 
-		imageStruct := objects.Image{Name: imgstr, Registry: url, Org: org, Manifest: manifest}
+		imageStruct := objects.Image{Name: imgstr, Registry: r.r.URL, Org: r.Org, Manifest: manifest}
```

Comment:
Redundant `r.` on `r.r.URL`?

Reply:
Not redundant, just poorly named.  The instance is named r and it has a member named r as well.